### PR TITLE
Fixed PlayerFishEvent documentation. Fixes BUKKIT-4975

### PR DESCRIPTION
**Pull request by fragtastic**
Method getCaught() actually returns an Item as Entity if fishing successful, not null.

### CODE REVIEW

**Comment by Wolvereness** [comment]
This doesn't follow proper formatting. A `<p>` tag should be used before secondary statements. A blank line should be maintained before the annotations. Lines should be wrapped to 78 characters (79th being \n).

File: src/main/java/org/bukkit/event/player/PlayerFishEvent.java [hunk 1]
```diff
@@ -38,8 +38,9 @@ public PlayerFishEvent(final Player player, final Entity entity, final Fish hook
 
     /**
      * Gets the entity caught by the player
-     *
-     * @return Entity caught by the player, null if fishing, bobber has gotten stuck in the ground or nothing has been caught
+     * <p>
+     * If player has fished successfully, the result may be cast to {@link Item}.
```
Comment (aerouk) [inline src/main/java/org/bukkit/event/player/PlayerFishEvent.java:42]: You'll need a line break here.

**Comment by aerouk** [comment]
Also, on line 43, you need to get rid of the space after the <code>*</code> symbol.

**Comment by Wolvereness** [comment]
This needs to be updated to the latest master.

Could you also provide _proof_ for the behavior defined? Links to craftbukkit should suffice.

/edit: And at least make a comment indicating that it has been updated, for alert reasons.

**Comment by fragtastic** [comment]
Proof.
https://github.com/Bukkit/CraftBukkit/blob/master/src/main/java/net/minecraft/server/EntityFishingHook.java

Lines 390 & 392.
EntityItem entityitem = new EntityItem(this.world, this.locX, this.locY, this.locZ, this.f());

PlayerFishEvent playerFishEvent = new PlayerFishEvent((Player) this.owner.getBukkitEntity(), entityitem.getBukkitEntity(), (Fish) this.getBukkitEntity(), PlayerFishEvent.State.CAUGHT_FISH);

**Comment by aerouk** [comment]
The doc on line 44 exceeds 78 characters.

File: src/main/java/org/bukkit/event/player/PlayerFishEvent.java [hunk 2]
```diff
@@ -37,10 +37,12 @@ public PlayerFishEvent(final Player player, final Entity entity, final Fish hook
...
 
     /**
-     * Gets the entity caught by the player
+     * Gets the entity caught by the player.
+     * <p>
+     * If player has fished successfully, the result may be cast to {@link Item}.
      *
-     * @return Entity caught by the player, null if fishing, bobber has gotten
-     *     stuck in the ground or nothing has been caught
+     * @return Entity caught by the player, Entity if fishing, and null if bobber
```
Comment (aerouk) [inline src/main/java/org/bukkit/event/player/PlayerFishEvent.java:44]: This line still exceeds the 78 character limit.

File: src/main/java/org/bukkit/event/player/PlayerFishEvent.java [hunk 3]
```diff
@@ -37,10 +37,12 @@ public PlayerFishEvent(final Player player, final Entity entity, final Fish hook
...
+     * has gotten stuck in the ground or nothing has been caught
```
Comment (aerouk) [inline src/main/java/org/bukkit/event/player/PlayerFishEvent.java:45]: You should indent wrapped lines by 4 spaces.

**Pull request merged**